doc/nrf/protocols/matter/end_product/security.rst

```diff
@@ -1,16 +1,121 @@
 .. _ug_matter_device_security:
 
-Securing production devices
-###########################
+Security
+########
 
 .. contents::
    :local:
-   :depth: 2
+   :depth: 3
+
+Nordic Matter samples leverage security features supported in the |NCS| that can be divided into three major categories:
+
+* Cryptography
+* Trusted storage
+* Securing production devices
+
+In the following sections you will learn more details about each listed category.
+
+Cryptography
+************
+
+Depending on the networking backend, the |NCS| Matter samples currently use the following APIs to implement cryptographic operations:
+
+* PSA Cryptography API for Thread networking.
+* Mbed TLS for Wi-Fi networking (it is planned to support PSA Cryptography API for Wi-Fi backend as well).
+
+Both APIs are integrated in the :ref:`nrf_security` library.
+This library offers various configuration possibilities and backends that can be employed to implement :ref:`cryptographic operations <cryptographic_operations_in_ncs>`.
+You can find an overview of the cryptography layer configuration supported for each |NCS| Matter-enabled platform in the :ref:`matter_platforms_security_support` section.
+
+Trusted storage
+***************
+
+:ref:`trusted_storage_in_ncs` is a security mechanism designed to securely store and manage sensitive data, such as cryptographic keys, device credentials, and configuration data.
+The mechanism is essential for IoT devices, as it allows the implementation of secure communication between devices.
+
+The Trusted Storage can utilize one of the following backends to implement PSA Certified Secure Storage API:
+
+* TF-M Platform Root of Trust (PRoT) - Only if the `ARM TrustZone`_ technology and hardware-accelerated firmware isolation is supported by the platform in use.
```

Comment:
```suggestion
* TF-M Platform Root of Trust (PRoT).
  This can only be utilized if the `ARM TrustZone`_ technology and hardware-accelerated firmware isolation are supported by the platform in use.
```